src/averageAges.js

```diff
@@ -55,10 +65,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filteredPeople = people
+    .filter(person => (onlyWithSon
+      ? person.sex === MALE_SEX
+      : true) && (people.some(child => child.name === person.mother)
+    ));
+
+  const ageAverage = filteredPeople.reduce((acc, { born, mother }) => {
+    const birthYearOfMom = people.find(({ name }) => name === mother).born;
+
+    return acc + born - birthYearOfMom;
+  }, 0);
+
+  return ageAverage / filteredPeople.length;
+}
+
+function getAvgAge(people) {
```

Comment:
Do not abbreviate names to unreadable ones.
```suggestion
function getAverageAge(people) {
```